### PR TITLE
Introduce CLI with flags

### PR DESCRIPTION
**Pull request by paulirish**
Starting simple. 
image

Next flags to add are `--save` and `--debug`

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var chalk = require('chalk');
```
Comment (paullewis) [inline cli.js]: make `const`s plzz

**Comment by paullewis** [comment]
Minor nit with the `var` => `const`, but other than that LGTM

**Comment by brendankenny** [comment]
LGTM toooo

**Comment by samccone** [comment]
:+1:

File: index.js [hunk 2]
```diff
@@ -44,13 +44,16 @@ const audits = [
...
-      console.log(err.stack);
-      throw err;
-    });
+module.exports = function(opts) {
+  var url = opts.url || defaultUrl;
+  Gatherer
+      .gather(gatherers, {url, driver})
+      .then(artifacts => Auditor.audit(artifacts, audits))
+      .then(results => {
+        console.log(results);
```
Comment (brendankenny) [inline index.js]: eventually will we want to pipe this somewhere, with the default being pipe it to console? Unless we want it to always be stdout and the user is in charge of piping

Comment (samccone) [inline index.js]: console.log is a direct proxy to stdout with a newline inserted at the end :goat:

Comment (paullewis) [inline index.js]: yeah I would expect:
```bash
lighthouse https://example.com | somethingelse
```

Comment (samccone) [inline index.js]: this should work :)

**Comment by paulirish** [comment]
note that this PR breaks "node ." 
so you'll have to `node cli.js` or `npm link` & `lighthouse`

File: readme.md [hunk 3]
```diff
@@ -16,19 +16,36 @@ Will run the audits defined in the `audits/` folder against a demo app. Chrome p
...
 ## Tests
 
-Some basic unit tests forked from node-big-rig are in `/test` and run via mocha. eslint is also checked for style violations.
+Some basic unit tests forked are in `/test` and run via mocha. eslint is also checked for style violations.
 
 ```js
 npm test
 ```
 
+### Code Style
```
Comment (paulirish) [inline readme.md]: FYI: I started some readme notes on code style. Let's try to document our decisions there.

**Comment by samccone** [comment]
chill :bread:

**Pull request merged**